### DISCUSSION TITLE
What center-app is expected to do, exactly?

### DISCUSSION THREAD

**Discussion by kenNash** [johnlindquist/kit#503] [Q&A] [answered]
Hi everyone!

I'm new here, so please forgive my sillinesses…

I'm trying the center-app script, but it does nothing for me. I don't even know what to expect, exactly. What does "centering an app" mean? Centering the frontmost window? All of the app windows?

I'd like to have scripts for moving/resizing the frontmost window in different places/positions. But so far I can't figure out…
how…

Any help would be very much appreciated.

--
kenNash

**Comment by johnlindquist** [comment, marked as answer]
It is supposed to move the frontmost app to the center of the screen.


"center-app" uses `setActiveAppBounds`. The implementation of `setActiveAppBounds` looks like this:

```js
global.setActiveAppBounds = async ({
  left,
  top,
  right,
  bottom,
}) => {
  await applescript(
    `tell application "System Events"
      set processName to name of first application process whose frontmost is true as text
      tell process processName to set the position of front window to {${left}, ${top}}
      tell process processName to set the size of front window to {${
        right - left
      }, ${bottom - top}}
    end tell`
  )
}
```

Notice the `System Events` part of the AppleScript. A `System Events` call will usually pop open a "Kit needs to permission to control System Events". Usually, if the script isn't doing anything, it's because that prompt was dismissed, so the script fails silently.

You can check to see if Kit has access to System Events here:

![CleanShot 2021-11-02 at 07 46 45](https://user-images.githubusercontent.com/36073/139859469-977ad923-f988-4ca9-a485-de0c8175ada9.png)

Does that help?

**Reply by kenNash** [reply]
Thank you John!

Unfortunately, this doesn't solve my problem. System Events.app is checked, but nothing happens…

**Reply by kenNash** [reply]
Uh, after checking Kit in the Accessibility pane, as well, it now works!

Thank you again for you kind help!

**Reply by johnlindquist** [reply]
Cool, glad you got it. Those pref settings are always a bit of a pain. Let me know if you have any more questions. Questions really help me improve the docs/guides that I'm working on!
